gen/build_deploy/bash.py

```diff
@@ -465,52 +465,7 @@ def validate_custom_check_bins_dir(custom_check_bins_dir):
 
     check_docker_running
 
-    local docker_version=$(command -v docker >/dev/null 2>&1 && docker version 2>/dev/null | awk '
-        BEGIN {
-            version = 0
-            client_version = 0
-            server_version = 0
-        }
-        {
-            if($1 == "Server:") {
-                server = 1
-                client = 0
-            } else if($1 == "Client:") {
-                server = 0
-                client = 1
-            } else if ($1 == "Server" && $2 == "version:") {
-                server_version = $3
-            } else if ($1 == "Client" && $2 == "version:") {
-                client_version = $3
-            }
-            if(server && $1 == "Version:") {
-                server_version = $2
-            } else if(client && $1 == "Version:") {
-                client_version = $2
-            }
-        }
-        END {
-            if(client_version == server_version) {
-                version = client_version
-            } else {
-                cv_length = split(client_version, cv, ".")
-                sv_length = split(server_version, sv, ".")
-
-                y = cv_length > sv_length ? cv_length : sv_length
-
-                for(i = 1; i <= y; i++) {
-                    if(cv[i] < sv[i]) {
-                        version = client_version
-                        break
-                    } else if(sv[i] < cv[i]) {
-                        version = server_version
-                        break
-                    }
-                }
-            }
-            print version
-        }
-    ')
+    local docker_version=$(command -v docker >/dev/null 2>&1 && docker version --format '\{\{printf "%s\n%s" .Server.Version .Client.Version\}\}' | sort -V | head -n 1)
```

Comment:
```suggestion
    local docker_version=$(command -v docker >/dev/null 2>&1 && docker version --format '\{\{printf "%s\n%s" .Server.Version .Client.Version\}\}' | sort -rV | head -n 1)
```
As far as I understand the old AWK script the lesser of both versions should be returned here.

Reply:
The code it good
```[root@dcos-e2e-default-4cc8d-master-0 /]# echo -e '18.06.3-ce\n17.06.3' | sort -V | head -n 1
17.06.3
[root@dcos-e2e-default-4cc8d-master-0 /]# echo -e '17.06.1\n18.06.3-ce\n17.06.3' | sort -V | head -n 1
17.06.1
```

Reply:
Aaah, sorry, you're using `head`, not `tail`. All good.